l1-contracts/src/core/messagebridge/Registry.sol

```diff
@@ -88,4 +93,36 @@ contract Registry is IRegistry {
   {
     return currentSnapshot;
   }
+
+  /**
+   * @notice Creates a new snapshot of the registry
+   * todo: this function must be permissioned with some kind of governance/voting/authority
+   * @param _rollup - The address of the rollup contract
+   * @param _inbox - The address of the inbox contract
+   * @param _outbox - The address of the outbox contract
+   * @return The version of the new snapshot
+   */
+  function upgrade(address _rollup, address _inbox, address _outbox)
+    public
+    override(IRegistry)
+    returns (uint256)
+  {
+    (, bool exists) = _getVersionFor(_rollup);
+    if (exists) revert Errors.Registry__RollupAlreadyRegistered(_rollup);
```

Comment:
can we just add a dumb test to ensure this never gets deleted

Reply:
Added a registry test file 👍